run.py

```diff
@@ -22,9 +28,71 @@ def main():
     logging.basicConfig(**log_format_config)
     logging.getLogger('requests.packages.urllib3.connectionpool').setLevel(
         logging.WARNING)
-    bot = Bot()
-    bot.run()
 
 
-if __name__ == '__main__':
-    main()
+def analyze_message(messages: List[dict]) -> Callable[[BaseClient], None]:
+    """コマンド解析"""
+
+    if len(messages) > 0 and messages[0]['type'] == 'text':
+        message = messages[0]['text'].strip()
+        if message.startswith('help'):
+            return hato.help_message
+        if message.startswith('eq') or message.startswith('地震'):
+            return hato.earth_quake
+        if message.startswith('in'):
+            return hato.labotter_in
+        if message.startswith('rida'):
+            return hato.labotter_rida
+        if message.startswith('text list'):
+            return hato.get_text_list
+        if message.startswith('text add '):
+            return hato.add_text(message[len('text add '):])
+        if message.startswith('text show '):
+            return hato.show_text(message[len('text show '):])
+        if message.startswith('text delete '):
+            return hato.delete_text(message[len('text delete '):])
+        if message.startswith('text random') or message.startswith('text'):
+            return hato.show_random_text
+        if message.startswith('天気'):
+            return hato.weather((message[len('天気'):]).strip())
+        if message.startswith('>< '):
+            return hato.totuzensi(message[len('>< '):])
+        if message == 'amesh':
+            return hato.amesh
+        if message.startswith('amesh '):
+            return hato.amesh_with_gis((message[len('amesh '):]).strip())
+        if message.startswith('version'):
+            return hato.version
+
+    return hato.default_action
+
+
+TPE = ThreadPoolExecutor(max_workers=3)
+
+
+@slack_events_adapter.on("app_mention")
+def on_app_mention(event_data):
+    """
+    appにメンションが送られたらここが呼ばれる
+    """
+
+    channel = event_data["event"]["channel"]
+    blocks = event_data['event']['blocks']
+    authed_users = event_data['authed_users']
+
+    for block in blocks:
+        if block['type'] == 'rich_text':
+            block_elements = block['elements']
+            for block_element in block_elements:
+                if block_element['type'] == 'rich_text_section':
+                    block_element_elements = block_element['elements']
+                    if len(block_element_elements) > 0 and \
+                            block_element_elements[0]['type'] == 'user' and \
+                    block_element_elements[0]['user_id'] in authed_users:
+                        TPE.submit(analyze_message(block_element_elements[1:]), SlackClient(
+                            channel, block_element_elements[0]['user_id']))
```

Comment:
mapもfilterもreduceも要らなかった。
とっとと早期continueすればよい by nakei
```suggestion
for block in blocks:
        if block['type'] != 'rich_text':
                continue
        block_elements = block['elements']
        for block_element in block_elements:
                if (block_element['type'] != 'rich_text_section'
                        or len(block_element_elements) == 0
                        or block_element_element[0]['type'] != 'user'
                        or block_element_elements[0]['user_id'] not in authed_users):
                    continue
                TPE.submit(
                    analyze_message(block_element_elements[1:]),
                    SlackClient(channel, block_element_elements[0]['user_id'])
                )
```

Reply:
ごめん、一段ネストして